clippy_lints/src/misc.rs

```diff
@@ -519,7 +519,9 @@ fn is_signum(cx: &LateContext<'_, '_>, expr: &Expr) -> bool {
 }
 
 fn is_float(cx: &LateContext<'_, '_>, expr: &Expr) -> bool {
-    matches!(walk_ptrs_ty(cx.tables.expr_ty(expr)).sty, ty::Float(_))
+    let value = &walk_ptrs_ty(cx.tables.expr_ty(expr)).sty;
+
+    matches!(value, ty::Array(arr_ty, _)) && matches!(value, ty::Float(_)) || matches!(value, ty::Float(_))
```

Comment:
```suggestion
    matches!(value, ty::Array(arr_ty, _)) && matches!(&walk_ptrs_ty(arr_ty).sty, ty::Float(_)) || matches!(value, ty::Float(_))
```

Reply:
Unfortunately, getting an error that:

```Rust
`arr_ty` is not found in this scope
```

Could it be an issue with the macro?

Reply:
In that case you have to write the match yourself. The macro seems to open a new scope for the variables.

Reply:
Hey. Was caught up this week, didn't get the chance to take a look at this further. 

Going to resume next week!